skimage/feature/masked_register_translation.py

```diff
@@ -0,0 +1,275 @@
+"""
+Implementation of the masked normalized cross-correlation (MNXC)
+
+Based on the following publication:
+D. Padfield. Masked object registration in the Fourier domain.
+IEEE Transactions on Image Processing (2012)
+
+and the author's original MATLAB implementation, available on this website:
+http://www.dirkpadfield.com/
+"""
+
+import numpy as np
+from functools import partial
+
+# next_fast_len was implemented in scipy 0.18
+# In case it cannot be imported, we use the id function
+# It could be much slower, but I don't think we should back-port
+# next_fast_length in its entirety
+# Take a look at the source:
+# https://github.com/scipy/scipy/blob/v1.1.0/scipy/fftpack/helper.py#L53-L151
+try:
+    from scipy.fftpack import next_fast_len
+except ImportError:
+    next_fast_len = lambda size: size
+
+
+def masked_register_translation(
+        src_image,
+        target_image,
+        src_mask,
+        target_mask=None,
+        overlap_ratio=3 / 10):
+    """
+    Masked image translation registration by masked normalized
+    cross-correlation.
+
+    Parameters
+    ----------
+    src_image : ndarray
+        Reference image.
+    target_image : ndarray
+        Image to register.  Must be same dimensionality as ``src_image``,
+        but not necessarily the same size.
+    src_mask : ndarray
+        Boolean mask for ``src_image``. The mask should evaluate to ``True``
+        (or 1) on valid pixels. ``src_mask`` should have the same shape
+        as ``src_mask``.
+    target_mask : ndarray or None, optional
+        Boolean mask for ``target_image``. The mask should evaluate to ``True``
+        (or 1) on valid pixels. ``target_mask`` should have the same shape
+        as ``target_image``. If ``None``, ``src_mask`` will be used.
+    overlap_ratio : float, optional
+        Maximum allowed overlap ratio between masks. The correlation at pixels
+        with an overlap ratio higher than this threshold will be zeroed. A 
+        larger percentage leads to greater robustness to spurious matches, for 
+        example due to noise. A smaller percentage leads to greater 
+        registration range.
+
+    Returns
+    -------
+    shifts : ndarray
+        Shift vector (in pixels) required to register ``target_image`` with
+        ``src_image``. Axis ordering is consistent with numpy (e.g. Z, Y, X)
+
+    References
+    ----------
+    .. [1] Dirk Padfield. Masked Object Registration in the Fourier Domain.
+        IEEE Transactions on Image Processing, vol.21(5), pp. 2706-2718 (2012).

```

Comment:
Please, make the citation consistent with the example. In particular, add the DOI.